pkg/prometheus/operator.go

```diff
@@ -83,6 +83,14 @@ func (labels *Labels) String() string {
 	return fmt.Sprintf(labels.LabelsString)
 }
 
+// Merge labels
+func (labels *Labels) Merge(dict map[string]string) map[string]string {
+	for key, value := range labels.LabelsMap {
```

Comment:
I would prefer if we create an entirely new map here and transfer both sets of key/value pairs into the new map and not modify any of the inputs.